control-plane/agents/core/src/node/service.rs

```diff
@@ -84,11 +84,11 @@ impl Service {
     /// Register a new node through the register information
     pub(super) async fn register(&self, registration: &Register) {
         self.registry.register_node_spec(registration).await;
-        self.register_state(registration).await;
+        self.register_state(registration, false).await;
     }
 
     /// Attempt to Register a new node state through the register information
-    pub(super) async fn register_state(&self, registration: &Register) {
+    pub(super) async fn register_state(&self, registration: &Register, startup: bool) {
```

Comment:
Could we do the initial state registration when we create the node service (i.e. in the `new` function)? We currently create the node service first and then immediately register the states.

Reply:
Yes indeed, done